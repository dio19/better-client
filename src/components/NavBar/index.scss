.navbar {
  display: flex;
  width: 100%;
  height: 70px;
  justify-content: space-between;
  align-items: center;
  position: relative;
  background-color: rgba(#000, 0.4);
  margin-bottom: 20px;

  &__list {
    display: flex;
    flex-direction: row;
    list-style: none;

    &-item {
      display: inline-block;
      margin: 15px;

      &-link {
        text-decoration: none;
        color: #fff;
      }

      &-link:hover {
        border-bottom: 5px solid #fd3838;
      }
    }
  }

  &__title {
    text-decoration: none;
    color: #fff;
    padding: 0 12px;
  }
}
