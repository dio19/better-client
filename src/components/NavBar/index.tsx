import React from "react";
import { NavLink } from "react-router-dom";
import "./index.scss";

const NavBar = () => {
  return (
    <nav className="navbar">
      <NavLink
        className="navbar__title"
        to="/"
        children={<h1>Customers App</h1>}
      ></NavLink>
      <ul className="navbar__list">
        <NavLink
          to="/"
          children={
            <li className="navbar__list-item">
              <div className="navbar__list-item-link">Home</div>
            </li>
          }
        ></NavLink>
        <NavLink
          to="/customers"
          children={
            <li className="navbar__list-item">
              <div className="navbar__list-item-link">List</div>
            </li>
          }
        ></NavLink>
        <NavLink
          to="/add-customer"
          children={
            <li className="navbar__list-item">
              <div className="navbar__list-item-link">Add</div>
            </li>
          }
        ></NavLink>
        <NavLink
          to="/update-or-delete"
          children={
            <li className="navbar__list-item">
              <div className="navbar__list-item-link">Update</div>
            </li>
          }
        ></NavLink>
      </ul>
    </nav>
  );
};

export default NavBar;
