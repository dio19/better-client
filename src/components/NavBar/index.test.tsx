import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import NavBar from "./index";

describe("NavBar Component", () => {
  test("It should render NavBar Component correctly", () => {
    const { asFragment } = render(
      <Router>
        <NavBar />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
