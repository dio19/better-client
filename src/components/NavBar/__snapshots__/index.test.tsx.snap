// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component It should render NavBar Component correctly 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <a
      aria-current="page"
      class="navbar__title active"
      href="/"
    >
      <h1>
        Customers App
      </h1>
    </a>
    <ul
      class="navbar__list"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <li
          class="navbar__list-item"
        >
          <div
            class="navbar__list-item-link"
          >
            Home
          </div>
        </li>
      </a>
      <a
        class=""
        href="/customers"
      >
        <li
          class="navbar__list-item"
        >
          <div
            class="navbar__list-item-link"
          >
            List
          </div>
        </li>
      </a>
      <a
        class=""
        href="/add-customer"
      >
        <li
          class="navbar__list-item"
        >
          <div
            class="navbar__list-item-link"
          >
            Add
          </div>
        </li>
      </a>
      <a
        class=""
        href="/update-or-delete"
      >
        <li
          class="navbar__list-item"
        >
          <div
            class="navbar__list-item-link"
          >
            Update
          </div>
        </li>
      </a>
    </ul>
  </nav>
</DocumentFragment>
`;
